src/payment/strategies/adyenv2/adyenv2-payment-strategy.ts

```diff
@@ -242,131 +162,111 @@ export default class AdyenV2PaymentStrategy implements PaymentStrategy {
-                    fingerprintToken: resultObject.token,
-                    onComplete: (fingerprintData: ThreeDS2OnComplete) => {
-                        const fingerprintPaymentPayload = {
-                            ...fingerprintData.data,
-                            paymentData: resultObject.payment_data,
-                        };
-
-                        const paymentPayload = {
-                            methodId: paymentMethodId,
-                            paymentData: {
-                                nonce: JSON.stringify(fingerprintPaymentPayload),
-                            },
-                        };
+                paymentComponent.mount(`#${adyenv2.containerId}`);
+
+                this._paymentComponent = paymentComponent;
 
-                        if (onComplete) {
-                            onComplete();
-                        }
+                if (adyenv2.cardVerificationContainerId) {
+                    const cardVerificationComponent = adyenCheckout.create(AdyenComponentType.SecuredFields, {
+                        ...adyenv2.options,
+                        onChange: componentState => this._updateComponentState(componentState),
+                        onError: componentState => this._updateComponentState(componentState),
+                    });
 
-                        resolve(paymentPayload);
+                    cardVerificationComponent.mount(`#${adyenv2.cardVerificationContainerId}`);
+
+                    this._cardVerificationComponent = cardVerificationComponent;
+                }
+                break;
+
+            case AdyenPaymentMethodType.AliPay:
+            case AdyenPaymentMethodType.Sofort:
+            case AdyenPaymentMethodType.WeChatPayQR:
+                this._updateComponentState({
+                    data: {
+                        paymentMethod: {
+                            type: paymentMethodName,
+                        },
                     },
-                    onError: (error: AdyenError) => reject(error),
                 });
+        }
+    }
 
-            const { threeDS2ContainerId, additionalActionOptions } = this._getAdyenV2PaymentInitializeOptions();
-
-            if (onBeforeLoad) {
-                onBeforeLoad();
-            }
+    private async _processAdditionalAction(error: any, shouldSaveInstrument?: boolean): Promise<any> {
+        if (!(error instanceof RequestError) || !some(error.body.errors, {code: 'additional_action_required'})) {
+            return Promise.reject(error);
+        }
 
-            fingerprintComponent.mount(`#${additionalActionOptions.containerId || threeDS2ContainerId}`);
+        const payment = await this._handleAction(error.body.provider_data);
 
-            if (onLoad) {
-                onLoad(() => {
-                    fingerprintComponent.unmount();
-                });
-            }
-        });
+        return await this._store.dispatch(this._paymentActionCreator.submitPayment({
```

Comment:
usually when you use await you don't use `.catch()` but instead `try { } catch() { }`
```ja
try {
  return await this._store.dispatch(this._paymentActionCreator.submitPayment({
} catch (error) {
  return this._processAdditionalAction(error, shouldSaveInstrument
}
```